subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/StaleDataCleanupInitializer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.results;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class StaleDataCleanupInitializer implements ConnectionAction<Void> {
+    private static final int MAX_DATA_STORE_DAYS = 365;
+
+    @Override
+    public Void execute(Connection connection) throws SQLException {
+        Statement statement = connection.createStatement();
+
+        String startDate = LocalDateTime.now().minusDays(MAX_DATA_STORE_DAYS).format(DateTimeFormatter.ISO_DATE);
+        statement.execute(String.format("delete from testExecution where startTime < '%s'", startDate));
+        statement.execute(String.format("delete from testOperation where testExecution in (select id from testExecution where startTime < '%s')", startDate));
```

Comment:
I think you need to delete from `testOperation` first and then from `testExecution`.

Moreover, don't use `String.format` with SQL statements, but prepared statements. There is no user code generating the SQL fields, though I still wouldn't use `String.format`.

Reply:
I noticed the order issue immediately after pushing.

Reply:
Prepared statement is used.